src/main/java/org/apache/lucene/analysis/kr/utils/DictionaryUtil.java

```diff
@@ -23,100 +23,71 @@
-        @Override
-        public Map<String, String> get() {
-            return readDict(KoreanEnv.FILE_PREFIX);
-        }
-    });
-
-    private static Supplier<Map<String, String>> SUFFIXS = Suppliers.memoize(new Supplier<Map<String, String>>() {
-        @Override
-        public Map<String, String> get() {
-            return readDict(KoreanEnv.FILE_SUFFIX);
-        }
-    });
-
-	private static Supplier<Map<String, WordEntry>> UNCOMPOUNDS = Suppliers.memoize(new Supplier<Map<String, WordEntry>>() {
-        @Override
-        public Map<String, WordEntry> get() {
-            Map<String, WordEntry> uncompounds = new HashMap<String, WordEntry>();
-            List<String> lines;
-
-            try {
-                lines = FileUtil.readLines(KoreanEnv.getInstance().getValue(KoreanEnv.FILE_UNCOMPOUNDS),"UTF-8");
-            } catch (Exception e) {
-                throw new RuntimeException("Failed to initialize UNCOMPOUNDS dictionary.", e);
-            }
-
-            for(String compound: lines) {
-                String[] infos = StringUtil.split(compound,":");
-                if(infos.length!=2) continue;
-                WordEntry entry = new WordEntry(infos[0].trim(),"90000X".toCharArray());
-                entry.setCompounds(compoundArrayToList(infos[1], StringUtil.split(infos[1],",")));
-                uncompounds.put(entry.getWord(), entry);
-            }
-
-            return Collections.unmodifiableMap(uncompounds);
-        }
-    });
-	
-	private static Supplier<Map<String, String>> CJWORDS = Suppliers.memoize(new Supplier<Map<String, String>>() {
-        @Override
-        public Map<String, String> get() {
-            Map<String, String> cjwords = new HashMap<String, String>();
-            List<String> lines;
-
-            try {
-                lines = FileUtil.readLines(KoreanEnv.getInstance().getValue(KoreanEnv.FILE_CJ), "UTF-8");
-            } catch (Exception e) {
-                throw new RuntimeException("Failed to initialize CJWORDS dictionary.", e);
-            }
-
-            for(String cj: lines) {
-                String[] infos = StringUtil.split(cj,":");
-                if(infos.length!=2) continue;
-                cjwords.put(infos[0], infos[1]);
-            }
-
-            return Collections.unmodifiableMap(cjwords);
-        }
-    });
+	private static Supplier<Map<String, String>> JOSAS = Memoizer.memoize(() -> readDict(KoreanEnv.FILE_JOSA));
 	
+	private static Supplier<Map<String, String>> EOMIS = Memoizer.memoize(() -> readDict(KoreanEnv.FILE_EOMI));
+
+    private static Supplier<Map<String, String>> PREFIXS = Memoizer.memoize(() -> readDict(KoreanEnv.FILE_PREFIX));
+
+    private static Supplier<Map<String, String>> SUFFIXS = Memoizer.memoize(() -> readDict(KoreanEnv.FILE_SUFFIX));
+
+	private static Supplier<Map<String, WordEntry>> UNCOMPOUNDS = Memoizer.memoize(() -> {
+		Map<String, WordEntry> uncompounds = new HashMap<>();
+		List<String> lines;
+
+		try {
```

Comment:
This whole file reading thing is ugly and could be easily done in "vanilla" Java, but I know you just changed indentation and it's not your bug. 😐